RESOURCE_ACCESS_CONTROL_FOR_PLUGINS.md

```diff
@@ -0,0 +1,511 @@
+| `resource_index`| String   | The OpenSearch index where the resource is stored. |
+
+#### **Response:**
+Returns whether the user has permission to access the resource.
+```json
+{
+  "has_permission": true
+}
+```
+
+#### **Response Fields:**
+| Field            | Type    | Description |
+|-----------------|---------|-------------|
+| `has_permission` | Boolean | `true` if the user has access, `false` otherwise. |
+
+---
+
+### **2. Grant Access**
+- **Endpoint:**
+  ```
+  POST /_plugins/_security/resources/share
+  ```
+- **Description:**
+  Grants access to a resource for specified **users, roles, and backend roles**.
+
+#### **Request Body:**
+```json
+{
+  "resource_id": "my-resource",
+  "resource_index": "resource-index",
+  "share_with": {
+      "users": ["shared-user-name"],
+      "roles": ["shared-roles"],
+      "backend_roles": ["shared-backend-roles"]
+  }
+}
+```
+
+#### **Request Fields:**
+| Field            | Type    | Description |
+|-----------------|---------|-------------|
+| `resource_id`   | String  | The unique identifier of the resource to be shared. |
+| `resource_index`| String  | The OpenSearch index where the resource is stored. |
+| `share_with`    | Object  | Defines which **users, roles, or backend roles** will gain access. |
+| `users`        | Array   | List of usernames allowed to access the resource. |
+| `roles`        | Array   | List of role names granted access. |
+| `backend_roles`| Array   | List of backend roles assigned to the resource. |
+
+#### **Response:**
+Returns the updated **resource sharing state**.
+```json
+{
+  "sharing_info": {
+    "source_idx": "resource-index",
+    "resource_id": "my-resource",
+    "created_by": {
+      "user": "you"
+    },
+    "share_with": {
+      "default": {
+        "users": ["shared-user-name"],
+        "roles": ["shared-roles"],
+        "backend_roles": ["shared-backend-roles"]
+      }
+    }
+  }
+}
+```
+
+Note: `default` is a place-holder action-group that will eventually be replaced with actual action-groups in the future.
+
+#### **Response Fields:**
+| Field          | Type    | Description |
+|---------------|---------|-------------|
+| `sharing_info` | Object  | Contains information about how the resource is shared. |
+| `source_idx`   | String  | The OpenSearch index containing the resource. |
+| `resource_id`  | String  | The unique identifier of the resource being shared. |
+| `created_by`   | Object  | Information about the user who created the sharing entry. |
+| `share_with`   | Object  | Defines users, roles, and backend roles with access to the resource. |
+
+---
+
+### **3. Revoke Access**
+- **Endpoint:**
+  ```
+  POST /_plugins/_security/resources/revoke
+  ```
+- **Description:**
+  Revokes access to a resource for specific users, roles, or backend roles.
+
+#### **Request Body:**
+```json
+{
+  "resource_id": "my-resource",
+  "resource_index": "resource-index",
+  "entities_to_revoke": {
+    "roles": ["shared-roles"]
+  }
```

Comment:
Does this need to be modified w.r.t. `default` action group?

Reply:
no, this is just a list of entities. Action-groups will be supplied separately (not in scope of v1)